### PR TITLE
No more hardcoded javadoc links

### PR DESCRIPTION
**Pull request by 12AwsomeMan34**
Resolves the ancient #127 when merged.

So to deal with the issue of hardcoding javadoc links, I've created a custom extension to suite our needs. This will allow the use of a `:javadoc:` role in the rst files. You may visit the [sphinx-JDLinker](https://github.com/12AwsomeMan34/sphinx-JDLinker) repository to view the extension.

The javadoc links are set in the `conf.py` file along with their root package. This is how sphinx-JDLinker knows which package belongs to which JavaDoc url.

Note that any docs editors will need to re-run `pip install -r etc/requirements.txt` once this is pulled in. You only need to do this once, unless the `sphinx-JDLinker` version is incremented.

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/index.html.

Here are some links to the pages that were modified:

- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/conf.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/contributing/implementation/datamanipulator.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/commands/arguments.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/commands/commandcallable.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/data/index.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/economy/basics.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/economy/using.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/3acf6dd0/plugin/manager.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/511#issuecomment-225777814), not to the direct url to the preview._

File: .gitignore [hunk 1]
```diff
@@ -131,3 +131,6 @@ sponge-theme/
 
 # Source code created by pip
 src/
+
+# compiled sponge-javadoc extension
+etc/sponge-javadoc/sponge-javadoc.pyc
```
Comment (Minecrell) [inline .gitignore]: You can just add `*.pyc` directly, they should never be uploaded to the docs repo

Comment (12AwsomeMan34) [inline .gitignore]: Hmm, seems its covered above by `*.py[cod]` anyway. Guess I'll just remove this one then.

File: source/plugin/blocks/modifying.rst [hunk 2]
```diff
@@ -96,17 +101,18 @@ The following example will dry the block at a given ``Location``, if possible.
...
+on block states that can not be wet since ``dryState`` will be ``Optional.empty()`` in that case.
 
 Copying Blocks
 ~~~~~~~~~~~~~~
 
-If you want to copy all of a blocks data, the ``BlockSnapshot`` class is your best friend. While it doesn't expose all
-the data, it stores a ``BlockType``, its ``BlockState`` and, if necessary, all additional Tile Entity Data (for example chest
-inventories). Conveniently, the ``Location`` class provides a ``createSnapshot()`` method to create a snapshot of the block
-at that point in time. That makes copying blocks from one location to another very simple:
+If you want to copy all of a blocks data, the :javadoc:`org.spongepowered.api.block.BlockSnapshot` class is your best
```
Comment (Inscrutable) [inline source/plugin/blocks/modifying.rst]: That should be "a block's data" - please use the [possessive apostrophe](http://www.thepunctuationguide.com/apostrophe.html) where appropriate.

Comment (12AwsomeMan34) [inline source/plugin/blocks/modifying.rst]: Heh, this one goes way back to https://github.com/SpongePowered/SpongeDocs/commit/cc84162d0b45ac060211f9ca2aeb28b84bdfc98c.

File: source/plugin/data/keys.rst [hunk 3]
```diff
@@ -87,18 +94,17 @@ Or, if you use Java 8, you're able to shorten the line with lambda expressions:
...
 Keyed Values
 ============
 
 There are cases where you may care about not only the direct value for a Key, but the keyed value
 encapsulating it. In that case, use the ``getValue(key)`` method instead of ``get(key)``. You will receive an
-object inheriting from :javadoc:`org.spongepowered.api.data.value.BaseValue` which contains a copy of the original
-value. Since we know that current health is a :javadoc:`org.spongepowered.api.data.value.mutable.MutableBoundedValue`,
-we can find out what is the minimum possible value and set our target's health just a tiny bit above that.
+object inheriting from :javadoc:`BaseValue` which contains a copy of the original value. Since we know that current
+health is a :javadoc:`MutableBoundedValue`, we can find out what is the minimum possible value and set our target's
```
Comment (Inscrutable) [inline source/plugin/data/keys.rst]: If you chop out "what is" it becomes more readable. (The "is" is in the wrong place anyway).

**Pull request merged**